hq/app/logic/CredentialsReportDisplay.scala

```diff
@@ -56,40 +57,34 @@ object CredentialsReportDisplay {
   }
 
   def toCredentialReportDisplay(report: IAMCredentialsReport): CredentialReportDisplay = {
+    val humanUsers = report.entries.filterNot(_.rootUser).collect {
+      case cred if cred.passwordEnabled.contains(true) =>
+        HumanUser(
+          cred.user,
+          cred.mfaActive,
+          accessKey1Details(cred),
+          accessKey2Details(cred),
+          humanReportStatus(cred),
+          dayDiff(lastActivityDate(cred)),
+          stack = cred.stack,
+          tags = cred.tags
+        )
+    }
 
-    report.entries.filterNot(_.rootUser).foldLeft(CredentialReportDisplay(report.generatedAt)) { (report, cred) =>
-      val machineUser =
-        if (!cred.passwordEnabled.getOrElse(false)) {
-          Some(MachineUser(
-            cred.user,
-            accessKey1Details(cred),
-            accessKey2Details(cred),
-            machineReportStatus(cred),
-            dayDiff(lastActivityDate(cred)),
-            stack = cred.stack,
-            tags = cred.tags
-          ))
-        } else None
-
-      val humanUser =
-        if (cred.passwordEnabled.getOrElse(false)) {
-          Some(HumanUser(
-            cred.user,
-            cred.mfaActive,
-            accessKey1Details(cred),
-            accessKey2Details(cred),
-            humanReportStatus(cred),
-            dayDiff(lastActivityDate(cred)),
-            stack = cred.stack,
-            tags = cred.tags
-          ))
-        } else None
-
-      report.copy(
-        machineUsers = report.machineUsers ++ machineUser,
-        humanUsers = report.humanUsers ++ humanUser
-      )
+    val machineUsers = report.entries.filterNot(_.rootUser).collect {
+      case cred if !cred.passwordEnabled.getOrElse(false) =>
```

Comment:
I'm confused that this conditional is different from the humanUsers one. Should this read `!cred.passwordEnabled.contains(true)`?

Reply:
Yes, good spot thank you. I think they're the same logically but I meant to update this one to match.